report/telegram.go

```diff
@@ -0,0 +1,83 @@
+package report
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/future-architect/vuls/config"
+	"github.com/future-architect/vuls/models"
+)
+
+// TelegramWriter sends report to Telegram
+type TelegramWriter struct{}
+
+func (w TelegramWriter) Write(rs ...models.ScanResult) (err error) {
+	conf := config.Conf.Telegram
+
+	for _, r := range rs {
+		serverInfo := fmt.Sprintf("%s", r.ServerInfo())
+		counter := 0
+		message := ""
+		for _, vinfo := range r.ScannedCves {
+			counter++
+			maxCvss := vinfo.MaxCvssScore()
+			severity := strings.ToUpper(maxCvss.Value.Severity)
+			if severity == "" {
+				severity = "?"
+			}
+
+			message += fmt.Sprintf(`*%s* \n [%s](https://nvd.nist.gov/vuln/detail/%s) _%s %s_\n %s\n\n`,
+				serverInfo,
+				vinfo.CveID,
+				vinfo.CveID,
+				strconv.FormatFloat(maxCvss.Value.Score, 'f', 1, 64),
+				severity,
+				vinfo.Summaries(config.Conf.Lang, r.Family)[0].Value)
+			if counter == 10 {
+				message = ""
+				if err = sendMessage(conf.Channel, conf.Token, message); err != nil {
+					return err
+				}
+			}
+		}
+		if message != "" {
+			if err = sendMessage(conf.Channel, conf.Token, message); err != nil {
+				return err
+			}
+		}
+
+	}
+	return nil
+}
+
+func sendMessage(channel, token, message string) error {
+	uri := fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", token)
+	payload := `{"text": "` + message + `", "chat_id": "@` + channel + `", "parse_mode": "Markdown" }`
```

Comment:
remove '@' of chat_id.
'@' is no need when sending a message to the user.
see https://stackoverflow.com/questions/31078710/how-to-obtain-telegram-chat-id-for-a-specific-user